sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/QuerySamples.cs

```diff
@@ -0,0 +1,235 @@
+
+            DateTimeOffset now = DateTime.UtcNow;
+            DateTimeOffset endTime = now.AddMinutes(10);
+            DateTimeOffset startTime = now.AddMinutes(-10);
+
+            var celsiusVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature"),
+                new TimeSeriesExpression("avg($value)"));
+            var fahrenheitVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature * 1.8 + 32"),
+                new TimeSeriesExpression("avg($value)"));
+
+            var querySeriesRequestOptions = new QuerySeriesRequestOptions();
+            querySeriesRequestOptions.InlineVariables["TemperatureInCelsius"] = celsiusVariable;
+            querySeriesRequestOptions.InlineVariables["TemperatureInFahrenheit"] = fahrenheitVariable;
+
+            QueryAnalyzer seriesQueryAnalyzer = client.Query.CreateSeriesQueryAnalyzer(
+                tsId,
+                startTime,
+                endTime,
+                querySeriesRequestOptions);
+
+            await foreach (TimeSeriesPoint point in seriesQueryAnalyzer.GetResultsAsync())
+            {
+                double? tempInCelsius = (double?)point.GetValue("TemperatureInCelsius");
+                double? tempInFahrenheit = (double?)point.GetValue("TemperatureInFahrenheit");
+
+                Console.WriteLine($"{point.Timestamp} - Average temperature in Celsius: {tempInCelsius}. Average temperature in Fahrenheit: {tempInFahrenheit}.");
+            }
+            #endregion Snippet:TimeSeriesInsightsSampleQuerySeries
+        }
+
+
+        private static async Task SendEventsToIotHubAsync(
+            DeviceClient deviceClient,
+            TimeSeriesId tsId,
+            TimeSeriesIdProperty[] timeSeriesIdProperties)
+        {
+            IDictionary<string, object> messageBase = BuildMessageBase(timeSeriesIdProperties, tsId);
+            double minTemperature = 20;
+            double minHumidity = 60;
+            var rand = new Random();
+
+            Console.WriteLine("\n\nSending temperature and humidity events to the IoT Hub.\n");
+
+            // Build the message base that is used as the base for every event going out
+            for (int i = 0; i < 10; i++)
+            {
+                double currentTemperature = minTemperature + rand.NextDouble() * 15;
+                double currentHumidity = minHumidity + rand.NextDouble() * 20;
+                messageBase["Temperature"] = currentTemperature;
+                messageBase["Humidity"] = currentHumidity;
+                string messageBody = JsonSerializer.Serialize(messageBase);
+                var message = new Message(Encoding.ASCII.GetBytes(messageBody))
+                {
+                    ContentType = "application/json",
+                    ContentEncoding = "utf-8",
+                };
+
+                await deviceClient.SendEventAsync(message).ConfigureAwait(false);
```

Comment:
Wondering if you have considered try catch here if in case if fails.

Reply:
I looked at the device samples to see what we do there. The samples do not wrap the send in a try/catch. I don't know why. Maybe to keep samples simple? I thought I would be consistent here as well.

Reply:
I also took a look. I do see reconnection sample following try catch practice. It's fine to keep it simple, this would be good to think about for future improvements/bring to parking lot.